src/components/PasswordField/PasswordField.vue

```diff
@@ -0,0 +1,318 @@
+		 * accessibility only. If you want to have the label visible just above
+		 * the input field pass in true to this prop.
+		 */
+		labelVisible: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * The placeholder of the input. This defaults as the string that's
+		 * passed into the label prop. In order to remove the placeholder,
+		 * pass in an empty string.
+		 */
+		placeholder: {
+			type: String,
+			default: undefined,
+		},
+
+		/**
+		 * Toggles the success state of the component. Adds a checkmark icon.
+		 * this cannot be used together with canClear.
+		 */
+		success: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggles the error state of the component. Adds an error icon.
+		 * this cannot be used together with canClear.
+		 */
+		error: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Additional error message
+		 *
+		 * This will be displayed beneath the input field
+		 */
+		helperText: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * The minlength property defines the minimum number of characters
+		 * (as UTF-16 code units) the user can enter
+		 */
+		minlength: {
+			type: Number,
+			default: 0,
+		},
+
+		/**
+		 * The maxlength property defines the maximum number of characters
+		 * (as UTF-16 code units) the user can enter
+		 */
+		maxlength: {
+			type: Number,
+			default: null,
+		},
+
+		/**
+		 * Helps the browser identify the type of password field and to provide
+		 * autocompletion for the password. Allowed values are off, on, new-password
+		 * old-password
+		 *
+		 * By default try to autocomplete with the current password
+		 */
+		autocomplete: {
+			type: String,
+			validator: (value) => [
+				'new-password',
+				'current-password',
+				'one-time-code',
+				'on',
+				'off',
+			].includes(value),
+			default: 'current-password',
+		},
+
+		/**
+		 * Check if the user entered a valid password using the password_policy
+		 * app if available.
+		 *
+		 * Warning: this doesn't replace server side checking and will do nothing
+		 * if the password_policy app is disabled.
+		 */
+		checkPasswordStrength: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Id of the input field. To use when using external label
+		 */
+		id: {
+			type: String,
+			default: '',
+		},
+	},
+
+	data() {
+		return {
+			isPasswordHidden: true,
+			internalErrorMessage: '',
+			passwordPolicy: null,
+		}
+	},
+
+	computed: {
+		computedHelperText() {
+			if (this.helperText.length > 0) {
+				return this.helperText
+			}
+			return this.internalErrorMessage
+		},
+		computedMinlength() {
+			if (this.$props.minlength) {
+				return this.$props.minlength
+			}
+			if (this.passwordPolicy) {
+				return this.passwordPolicy.minLength
+			}
+			return null
+		},
+	},
+
+	watch: {
+		value(newValue) {
+			if (this.checkPasswordStrength) {
+				if (this.passwordPolicy === null) {
+					this.passwordPolicy = loadState('core', 'capabilities').password_policy
```

Comment:
We can "preload" this value and avoid "recalculate" it every time.
Just move it to outside of component.

```js
const defaultPasswordPolicy = loadState('core', 'capabilities').password_policy

// -- 

this.passwordPolicy = defaultPasswordPolicy
```